### PR TITLE
feat: add custom search modal

### LINKED ISSUE #530: Search on stdlib page
Hi!

Please consider a search feature on the stdlib page - https://deno.land/std

deno’s stdlib are pretty big, and it includes some niceties e.g code examples. I think it’ll help people to explore and understand the stdlib by using the search feature.

### PR DESCRIPTION
**Pull request by crowlKats**
Closes #530

### CODE REVIEW

**Comment by ije** [comment]
the search modal will show automatically when open the the module docs page

**Comment by crowlKats** [comment]
oh whoops some debugging that was left on

File: islands/GlobalSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+  "Classes": "class",
+  "Enums": "enum",
+  "Variables": "variable",
+  "Functions": "function",
+  "Interfaces": "interface",
+  "Type Aliases": "typeAlias",
+} as const;
+
+/** Search Deno documentation, symbols, or modules. */
+export default function GlobalSearch() {
```
Comment (ije) [inline islands/GlobalSearch.tsx:41]: better lock the page scrolling when the modal shows

File: islands/GlobalSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+              <div
+                class={tw`lg:hidden ml-3 -mr-2 flex items-center`}
+                onClick={() => setShowModal(false)}
+              >
+                <Icons.Cross />
+              </div>
+            </div>
+
+            <div class={tw`flex gap-3 mt-2`}>
+              {kinds.map((k) => (
```
[the remark this answers is not in the document]
Comment (ije) [reply islands/GlobalSearch.tsx:182]: Screen Shot 2022-07-14 at 21 29 38

File: islands/GlobalSearch.tsx [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+          <span class={tw`font-semibold`}>{doc.name}</span>
+          <span class={tw`italic text-sm text-[#9CA0AA] leading-6`}>from</span>
+          <span>{location}</span>
+        </div>
+        {doc.jsDoc?.doc && (
+          <div
+            class={tw
+              `text-sm text-[#6C6E78] h-5 overflow-ellipsis overflow-hidden mr-24`}
+          >
+            {doc.jsDoc!.doc}
```
Comment (ije) [inline islands/GlobalSearch.tsx:318]: i think we just need to show the first line by `doc.jsDoc!.doc.split('\n')[0]`

**Review by ije — DISMISSED** [review]
nice job! please check the comments i leaved

File: util/twind.ts [hunk 4]
```diff
@@ -76,12 +78,17 @@ if (IS_BROWSER) {
...
     init(cb) {
       return cb(state.shift());
     },
   };
   config.sheet = sheet;
-  setup(config);
+  twSetup(config);
+  setup({
+    runtime: { Fragment, h },
+  });
```
Comment (lucacasonato) [inline util/twind.ts:93]:
```suggestion
  setup({
    runtime: { Fragment, h },
  });
```

We can not ship the entirety of doc components to the client.

File: islands/GlobalSearch.tsx [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+                class={tw
+                  `pl-4 h-10 w-full flex-shrink-1 bg-[#F3F3F3] rounded-md flex items-center text-light`}
+              >
+                <Icons.MagnifyingGlass />
+                <input
+                  class={tw`ml-1.5 py-3 leading-4 bg-transparent w-full`}
+                  type="text"
+                  onInput={(e) => setInput(e.currentTarget.value)}
+                  value={input}
+                  placeholder="Search manual, symbols and modules..."
```
[the remark this answers is not in the document]
Comment (crowlKats) [reply islands/GlobalSearch.tsx:169]: `autoFocus` doesnt seem to work
can you show me an example of `group`? am unsure

[the remark this answers is not in the document]
Comment (ije) [reply islands/GlobalSearch.tsx:170]: https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state

**Comment by crowlKats** [comment]
the issue is a comrak import

**Review by ry — APPROVED** [review]
LGTM

**Pull request merged**